models/articles.js

```diff
@@ -1,7 +1,47 @@
 const pool = require("../db/connection");
 
+module.exports.fetchAllArticles = (topic) => {
+	const queryParams = [];
+	let baseQuery = `
+			SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes,
+			COUNT (comments.article_id)::INT AS comment_count
+			FROM
+			comments
+			RIGHT JOIN articles
+			ON
+			articles.article_id = comments.article_id
+			`;
+	if (topic) {
+		queryParams.push(topic);
+		baseQuery += `
+		WHERE topic LIKE $1
+		GROUP BY (articles.article_id)
+		ORDER BY articles.created_at DESC;
+		`;
+	} else {
+		baseQuery += `
+		GROUP BY (articles.article_id)
+		ORDER BY articles.created_at DESC;
+		`;
```

Comment:
This part can be there by default and your if only adds the WHERE condition so you don't repeat the GROUP BY